crates/red_knot_python_semantic/resources/mdtest/comparison/instances/rich_comparison.md

```diff
@@ -315,3 +315,27 @@ def f(x: bool, y: int):
     reveal_type(4.2 < x)  # revealed: bool
     reveal_type(x < 4.2)  # revealed: bool
 ```
+
+## Instance Comparisons with not-boolable types
+
+<!-- snapshot-diagnostics -->
+
+Python implicitly calls `bool` on the comparison result of preceeding elements (but not for the last
+element).
+
+```py
+class NotBoolable:
+    __bool__ = 3
+
+class Comparable:
+    def __lt__(self, item) -> NotBoolable:
+        return NotBoolable()
+
+    def __gt__(self, item) -> NotBoolable:
+        return NotBoolable()
+
+# error: [not-boolable]
+10 < Comparable() < 20
+# error: [not-boolable]
+10 < Comparable() < Comparable()
```

Comment:
Could we also test that we _don't_ emit a diagnostic if there are only two elements?

```suggestion
10 < Comparable() < Comparable()

Comparable() < Comparable()  # fine
```